continuous-delivery/build-wheels-manylinux2014-x86_64-jenkins.sh

```diff
@@ -8,8 +8,11 @@ $(aws --region us-east-1 ecr get-login --no-include-email)
 
 docker pull $DOCKER_IMAGE
 
+# NOTE: run as current user to avoid git "dubious ownership" error,
+# and so that output artifacts don't belong to "root"
 docker run --rm \
     --mount type=bind,source=`pwd`,target=/aws-crt-python \
+    --user "$(id -u):$(id -g)" \
```

Comment:
I don't get why using current user will work. 

Isn't that because one jenkins job polls the repo and upload to S3? And the build runs on other jenkins job?

Are those two jekins job running on the same EC2, so it's the same user? Or they have the same user name and group name and git only checks for string match?

Reply:
I think the problem here was that the we build within a docker container on the Jenkins. The docker was under root privilege by default so it failed to access the git source owns by the user here.

Reply:
Turns out, when we unzip the file from S3, ownership of those files/folder is properly reassigned to the user. Git has no issue runnin on the unzipped files. The issue comes up when we MOUNT that unzipped folder into a docker container.

By default, inside a docker container, things run as `root`

But file ownership within that mounted folder is bananas. See:
```
[root@a8e45e5d0076 aws-crt-python]# ls -al
drwxr-xr-x 14 1778205 users  4096 Oct 26 15:56 .
drwxr-xr-x  1 root    root   4096 Oct 27 18:32 ..
drwxr-xr-x  9 1778205 users  4096 Oct 27 18:25 .git
drwxr-xr-x  3 1778205 users  4096 Oct 26 15:56 awscrt
...
```

Note how the parent directory is owned by `root`, but everything inside the mounted directory is owned by the user-id/group-id from host (the docker container doesn't know how to map the numeric user-id-num to a username, so it's just showing the number)

With git's new rules, when the `root` user runs git and see that the `.git/` folder is owned by someone else, it fails

I should also note that this behavior is inconsistent. When mounting a folder from my Mac, the docker container showed everything as being owned by `root` and git had no issues. The issue only occurred when mounting a folder from Amazon Linux machines, the ownership was NOT reassigned to `root`. I have no idea how inconsistent the behavior might be between different Linux distros? TBD?

We'll probably keep hitting this issue in the future, check out "Other fixes considered" in my writeup up top. This was the simplest change that didn't sacrifice security, so I went with it.